src/io/flutter/logging/FlutterLog.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.logging;
+
+import com.intellij.execution.process.ProcessAdapter;
+import com.intellij.execution.process.ProcessEvent;
+import com.intellij.execution.process.ProcessHandler;
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.util.Disposer;
+import com.intellij.openapi.util.Key;
+import com.jetbrains.lang.dart.ide.runner.server.vmService.VmServiceConsumers;
+import io.flutter.utils.VmServiceListenerAdapter;
+import org.dartlang.vm.service.VmService;
+import org.dartlang.vm.service.element.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FlutterLog {
+
+  // (Temporary) flag to enable logging support.
+  public static final boolean LOGGING_ENABLED = false;
+  public static final String LOGGING_STREAM_ID = "_Logging";
+
+  private static final FlutterLog INSTANCE = new FlutterLog();
+
+  private final List<Listener> listeners = new ArrayList<>();
+
+  // Shared instance.
+  private FlutterLog() {
+  }
+
+  public static FlutterLog getInstance() {
+    return INSTANCE;
+  }
+
+  public void addListener(@NotNull Listener listener, @Nullable Disposable parent) {
+    listeners.add(listener);
+    if (parent != null) {
+      Disposer.register(parent, () -> listeners.remove(listener));
+    }
+  }
+
+  public void removeListener(@NotNull Listener listener) {
+    listeners.remove(listener);
+  }
+
+  private void onEntry(@Nullable FlutterLogEntry entry) {
+    if (entry != null) {
+      for (Listener listener : listeners) {
+        listener.onEvent(entry);
+      }
+    }
+  }
+
+  public void listenToProcess(@NotNull ProcessHandler processHandler, @NotNull Disposable parent) {
```

Comment:
We probably want to invert things here, so that the FlutterLog doesn't know how to listen to n different things, but other places translate and log events to it (service protocol log messages, flutter frame events, ...).

Reply:
Another benefit is centralizing disposal logic.